src/annotation.ts

```diff
@@ -146,8 +149,12 @@ export class Annotation extends Modifier {
         annotation.setTextLine(state.text_line);
       }
     }
-    state.left_shift += width / 2;
-    state.right_shift += width / 2;
+    const overlap = Math.min(
+      Math.max(width - maxGlyphWidth, 0),
+      Math.max(width - (state.left_shift + state.right_shift), 0)
+    );
+    state.left_shift += overlap / 2;
+    state.right_shift += overlap / 2;
```

Comment:
I just realised that this calculations are ignoring `this.horizontalJustification`, are not day? Also shifted notes. The second problem affects articulations and accidentals as well.

Reply:
Of course the previuos ones were also ignoring that.

Reply:
You are right.  The horizontal formatting has never considered horizontal justification, and none of our test cases exercise this.  

I didn't think about the left and right shift being different, but that certainly could be.  We could consider a left and right Max in the for loop, and then a left and right overlap.  That should reduce to the same thing in the case of being centered. 2x as many calculations but the same logic.